src/gen-ts.ts

```diff
@@ -234,6 +245,63 @@ function handleFunction(feature: AnalyzerFunction, root: ts.Document) {
 }
 
 /**
+ * Convert the given Analyzer properties to their TypeScript declaration
+ * equivalent.
+ */
+function handleProperties(analyzerProperties: Iterable<analyzer.Property>):
+    ts.Property[] {
+  const tsProperties = new Array<ts.Property>();
+  for (const property of analyzerProperties) {
+    if (property.inheritedFrom) {
+      continue;
+    }
+    tsProperties.push({
+      kind: 'property',
+      name: property.name || '',
+      description: property.description || '',
+      type: property.type ? closureTypeToTypeScript(property.type) : ts.anyType,
+    });
+  }
+  return tsProperties;
+}
+
+
+/**
+ * Convert the given Analyzer methods to their TypeScript declaration
+ * equivalent.
+ */
+function handleMethods(analyzerMethods: Iterable<analyzer.Method>):
+    ts.Method[] {
+  const tsMethods = new Array<ts.Method>();
+  for (const method of analyzerMethods) {
+    if (method.inheritedFrom) {
+      continue;
+    }
+    const params: ts.Param[] = [];
+    for (const param of method.params || []) {
+      const {optional, type} = closureParamToTypeScript(param.type || '');
```

Comment:
the `||` could go away if `closureParamToTypeScript` handles null/undefined

Reply:
Done.